File: Mobile/Android/PLTDevice/src/com/plantronics/device/Subscription.java
package com.plantronics.device;

/**
 * Created by mdavis on 3/20/14.
 */
public class Subscription {

	private short 		_service;
	private byte 		_mode;
	private short		_period;


	public Subscription(short service, byte mode, short period) {
		_service = service;
		_mode = mode;
		_period = period;
	}

	public short getService() {
		return _service;
	}

	public byte getMode() {
		return _mode;
	}

	public short getPeriod() {
		return _period;
	}

	@Override
	public String toString() {
		return getClass().getName() + ": service=" + _service + ", mode=" + _mode + ", period=" + _period;
	}
}

File: Desktop/OS X/PLTDevice/PLTDevice/Subscription.swift
//
//  Subscription.swift
//  PLTDevice
//
//  Created by Morgan Davis on 6/3/14.
//  Copyright (c) 2014 Plantronics. All rights reserved.
//

import Foundation


class Subscription {
	
	// MARK: Properties
	
	//var service:		Device.Service
	//var mode:			Device.SubscriptionMode
	//var	period:			UInt16
	
	// MARK: Initializers
	
	init() {
//		self.service = Device.Service.OrientationTracking
//		self.mode = Device.SubscriptionMode.OnChange
//		self.period = 0
	}
	
	// PRIVATE ?!?!?!?!
	
	//var	subscribers:	NSMutableArray =		[]
	
//	convenience init(service: Device.Service, mode: Device.SubscriptionMode, period: UInt16, subscriber: DeviceSubscriber!) {
//		self.init()
//		self.service = service
//		self.mode = mode
//		self.period = period
//		self.subscribers.addObject(subscriber)
//	}
	
	func addSubscriber(subscriber: DeviceSubscriber!) {
//		removeSubscriber(subscriber)
//		if (!self.subscribers.containsObject(subscriber)) {
//			self.subscribers.addObject(subscriber);
//		}
	}
	
	func removeSubscriber(subscriber: DeviceSubscriber!) {
		//self.subscribers.removeObject(subscriber)
	}
	
	func privateDescription() -> String {
		return ""
	}
}
